app/javascript/components/public_health/CustomExport.js

```diff
@@ -436,32 +437,47 @@ class CustomExport extends React.Component {
                         <FontAwesomeIcon className="mr-1" icon={['fas', 'trash']} />
                         Delete
                       </Button>
-                      <Button
-                        id="custom-export-action-update"
-                        size="sm"
-                        variant="primary"
-                        disabled={!this.state.preset?.id}
-                        className="ml-1"
-                        style={{ outline: 'none', boxShadow: 'none' }}
-                        onClick={this.update}>
-                        <FontAwesomeIcon className="mr-1" icon={['fas', 'pen-alt']} />
-                        Update
-                      </Button>
+                      {this.state.preset?.name === '' ? (
+                        <OverlayTrigger overlay={<Tooltip>Please indicate a name for the saved Custom Export</Tooltip>}>
+                          <span>
+                            <Button size="sm" variant="primary" disabled className="ml-1" style={{ outline: 'none', boxShadow: 'none' }}>
+                              <FontAwesomeIcon className="mr-1" icon={['fas', 'pen-alt']} />
+                              Update
+                            </Button>
+                          </span>
+                        </OverlayTrigger>
+                      ) : (
+                        <Button
+                          id="custom-export-action-update"
+                          size="sm"
+                          variant="primary"
+                          className="ml-1"
+                          style={{ outline: 'none', boxShadow: 'none' }}
+                          onClick={this.update}>
+                          <FontAwesomeIcon className="mr-1" icon={['fas', 'pen-alt']} />
+                          Update
+                        </Button>
+                      )}
                     </React.Fragment>
-                  ) : this.state.preset?.name === '' ? (
-                    <OverlayTrigger overlay={<Tooltip>Please indicate a name for the saved Custom Export</Tooltip>}>
-                      <span>
-                        <Button size="sm" variant="primary" disabled style={{ outline: 'none', boxShadow: 'none', pointerEvents: 'none' }}>
+                  )}
+                  {!this.state.preset?.id && (
+                    <React.Fragment>
+                      {this.state.preset?.name === '' ? (
```

Comment:
I don't think this check and the "else" result are necessary here. Because they both just do the same thing (show the "save" button).

Reply:
not sure if I'm misinterpreting or missing something, butt the purpose of this check is to prevent users from hitting the Save button if the preset name is blank. One of these shows a disabled save button and the other shows the actual save button

Reply:
Nope, I was just missing something myself. Ignore this - resolved!